platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/DataFileCompactor.java

```diff
@@ -0,0 +1,483 @@
+        if (filesToCompact.isEmpty()) {
+            logger.debug(MERKLE_DB.getMarker(), "[{}] No need to compact, as the compaction plan is empty", storeName);
+            return false;
+        }
+
+        final int filesCount = filesToCompact.size();
+        logger.info(MERKLE_DB.getMarker(), "[{}] Starting compaction", storeName);
+
+        int highestExistingCompactionLevel =
+                filesToCompact.get(filesCount - 1).getMetadata().getCompactionLevel();
+        // target compaction level should  not exceed `maxCompactionLevel` config parameter
+        final int targetCompactionLevel = Math.min(highestExistingCompactionLevel + 1, config.maxCompactionLevel());
+
+        final long start = System.currentTimeMillis();
+
+        final long filesToCompactSize = getSizeOfFiles(filesToCompact);
+        logger.debug(
+                MERKLE_DB.getMarker(),
+                "[{}] Starting merging {} files / {}",
+                storeName,
+                filesCount,
+                formatSizeBytes(filesToCompactSize));
+
+        final List<Path> newFilesCreated = compactFiles(index, filesToCompact, targetCompactionLevel);
+
+        final long end = System.currentTimeMillis();
+        final long tookMillis = end - start;
+        if (reportDurationMetricFunction != null) {
+            reportDurationMetricFunction.accept(targetCompactionLevel, tookMillis);
+        }
+
+        final long compactedFilesSize = getSizeOfFilesByPath(newFilesCreated);
+        if (reportSavedSpaceMetricFunction != null) {
+            reportSavedSpaceMetricFunction.accept(
+                    targetCompactionLevel,
+                    (filesToCompactSize - compactedFilesSize) * UnitConstants.BYTES_TO_MEBIBYTES);
+        }
+
+        logCompactStats(
+                storeName,
+                tookMillis,
+                filesToCompact,
+                filesToCompactSize,
+                newFilesCreated,
+                targetCompactionLevel,
+                dataFileCollection);
+        logger.warn(
+                MERKLE_DB.getMarker(),
+                "[{}] Finished compaction {} files / {} in {} ms",
+                storeName,
+                filesCount,
+                formatSizeBytes(filesToCompactSize),
+                tookMillis);
+
+        return true;
+    }
+
+    /**
+     * This function creates a compaction plan (a set of files to be compacted). The plan is organized by compaction levels
+     * in ascending order. If there are not enough files to compact, then no files are compacted and the plan will be empty.
+     * If the current level doesn't reach minminNumberOfFilesToCompact threshold,
+     * then this level and the levels above it are not included in the plan.
+     * @return filter creating a compaction plan
+     */
+    public static UnaryOperator<List<DataFileReader<?>>> compactionPlan(int minNumberOfFilesToCompact) {
+
+        return dataFileReaders -> {
```

Comment:
Why a lambda? Extra complexity that adds no value.

Reply:
:) it was a lambda in the first place. Yes, in this particular case it can be replaced with a regular method